### PR TITLE
[IMP]  account: track lock dates changes

### PR DESCRIPTION
**Pull request by jbw-odoo**
Allow log date changes tracking by adding chatter on company and tracking lock date fields.

Task: 2651834
https://github.com/odoo/enterprise/pull/22479

### CODE REVIEW

File: addons/account/models/company.py [hunk 1]
```diff
@@ -36,14 +36,26 @@
...
     fiscalyear_last_day = fields.Integer(default=31, required=True)
     fiscalyear_last_month = fields.Selection(MONTH_SELECTION, default='12', required=True)
-    period_lock_date = fields.Date(string="Lock Date for Non-Advisers", help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this date. Use it for period locking inside an open fiscal year, for example.")
-    fiscalyear_lock_date = fields.Date(string="Lock Date", help="No users, including Advisers, can edit accounts prior to and inclusive of this date. Use it for fiscal year locking for example.")
-    tax_lock_date = fields.Date("Tax Lock Date", help="No users can edit journal entries related to a tax prior and inclusive of this date.")
+    period_lock_date = fields.Date(
+        string="Journals Entries Lock Date",
+        tracking=True,
+        help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this"
+        " date. Use it for period locking inside an open fiscal year, for example.")
```
Comment (smetl) [inline addons/account/models/company.py:49]: plz indent:
```
period_lock_date = fields.Date(
    string="Journals Entries Lock Date",
    tracking=True,
    help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this "
         "date. Use it for period locking inside an open fiscal year, for example.",
)
```

File: addons/account/models/company.py [hunk 2]
```diff
@@ -36,14 +36,26 @@
 
 
 class ResCompany(models.Model):
-    _inherit = "res.company"
+    _name = "res.company"
+    _inherit = ["res.company", "mail.thread"]
```
Comment (smetl) [inline addons/account/models/company.py:40]: I don't think this should be added by the `account` module. To be discussed

**Review by smetl — CHANGES_REQUESTED** [review]

File: addons/account/models/company.py (hunk 2 above)
Comment (smetl) [reply addons/account/models/company.py:40]: OK let's keep that on the account module

File: addons/account/models/company.py (hunk 1 above)
Comment (jbw-odoo) [reply addons/account/models/company.py]: done

**Review by smetl — APPROVED** [review]
@robodoo r+

**Comment by smetl** [comment]
@robodoo retry

**Comment by smetl** [comment]
@jbw-odoo The error seems related to your PR after all... can you have a look plz?

**Comment by jbw-odoo** [comment]
> @jbw-odoo The error seems related to your PR after all... can you have a look plz?

will do

**Comment by jbw-odoo** [comment]
added enterprise PR https://github.com/odoo/enterprise/pull/22479

**Comment by smetl** [comment]
@robodoo retry

**Comment by jbw-odoo** [comment]
> @robodoo retry

Maybe need to r+ ?

**Comment by smetl** [comment]
@robodoo r+

**Pull request closed without merging**